designer/server/src/common/helpers/auth/get-user-session.js

```diff
@@ -32,28 +31,37 @@ export async function getUserSession(request, session) {
 
 /**
  * @param {AuthCredentials | null} [credentials]
+ * @returns {credentials is AuthWithTokens}
  */
-export function getUserClaims(credentials) {
-  if (!credentials?.token) {
-    return
-  }
+export function hasAuthenticated(credentials) {
+  return !!(
+    credentials?.token &&
+    credentials.idToken &&
+    credentials.refreshToken
+  )
+}
 
+/**
+ * @param {AuthWithTokens} credentials
+ */
+export function getUserClaims(credentials) {
   const { decoded } = /** @type {UserToken<UserProfile>} */ (
     token.decode(credentials.token)
   )
 
+  if (!decoded.payload) {
+    throw new Error('Failed to decode access token')
+  }
+
   return decoded.payload
 }
 
 /**
  * @param {AuthCredentials | null} [credentials]
+ * @returns {credentials is AuthSignedIn}
```

Comment:
Should `hasUser()` return true, we narrow `AuthCredentials` to `AuthSignedIn`

This stops us needing `credentials?.user?.id` or other type checks